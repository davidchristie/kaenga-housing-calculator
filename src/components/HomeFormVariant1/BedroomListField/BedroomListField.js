import React, { Component, PropTypes } from 'react'
import { Button, ListGroup } from 'react-bootstrap'

import BedroomField from './BedroomField'

class BedroomListField extends Component {
  render () {
    const { fields } = this.props
    return (
      <div>
        <Button onClick={() => fields.push({})}>
          Add Bedroom
        </Button>
        <ListGroup>
          {fields.map((bedroom, index) =>
            <BedroomField
              bedroom={bedroom}
              index={index}
              key={index}
              remove={() => fields.remove(index)}
            />
          )}
        </ListGroup>
      </div>
    )
  }
}

BedroomListField.defaultProps = {
  fields: {
    map: () => null
  }
}

BedroomListField.propTypes = {
  fields: PropTypes.shape({
    map: PropTypes.func.isRequired
  }).isRequired
}

export default BedroomListField
