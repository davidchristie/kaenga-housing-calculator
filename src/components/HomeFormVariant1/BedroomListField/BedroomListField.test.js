import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import BedroomListField from './BedroomListField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(
    shallowRenderer.render(<BedroomListField />)
  ).toMatchSnapshot()
})
