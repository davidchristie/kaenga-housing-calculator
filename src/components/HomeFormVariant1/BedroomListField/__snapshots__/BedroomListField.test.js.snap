exports[`test matches snapshot 1`] = `
<div>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}>
    Add Bedroom
  </Button>
  <ListGroup
    bsClass="list-group" />
</div>
`;
